amd_openvx_extensions/amd_rpp/source/tensor/Resample.cpp

```
@@ -0,0 +1,273 @@
+    RpptDescPtr pDstDesc;
+    Rpp32s *pSampleFrames;
+    Rpp32s *pSampleChannels;
+    Rpp32f *pInRateTensor;
+    Rpp32f *pOutRateTensor;
+    RpptResamplingWindow window;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+inline float sinc(float x) {
+    x *= M_PI;
+    return (std::abs(x) < 1e-5f) ? (1.0f - x * x * (1.0f / 6)) : std::sin(x) / x;
+}
+
+inline double hann(double x) {
+    return 0.5 * (1 + std::cos(x * M_PI));
+}
+
+// initialization function used for filling the values in Resampling window (RpptResamplingWindow)
+// using the coeffs and lobes value this function generates a LUT (look up table) which is further used in Resample audio augmentation
+inline void windowed_sinc(RpptResamplingWindow &window, int coeffs, int lobes) {
+    float scale = 2.0f * lobes / (coeffs - 1);
+    float scale_envelope = 2.0f / coeffs;
+    window.coeffs = coeffs;
+    window.lobes = lobes;
+    window.lookup.clear();
+    window.lookup.resize(coeffs + 5);
+    window.lookupSize = window.lookup.size();
+    int center = (coeffs - 1) * 0.5f;
+    for (int i = 0; i < coeffs; i++) {
+        float x = (i - center) * scale;
+        float y = (i - center) * scale_envelope;
+        float w = sinc(x) * hann(y);
+        window.lookup[i + 1] = w;
+    }
+    window.center = center + 1;
+    window.scale = 1 / scale;
+    window.pCenter = _mm_set1_ps(window.center);
+    window.pScale = _mm_set1_ps(window.scale);
+}
+
+void update_destination_roi(ResampleLocalData *data, RpptROI *src_roi, RpptROI *dst_roi) {
+    float scale_ratio;
+    for (unsigned i = 0; i < data->pSrcDesc->n; i++) {
+        data->pSampleFrames[i] = src_roi[i].xywhROI.roiWidth;
+        data->pSampleChannels[i] = src_roi[i].xywhROI.roiHeight;
+        if (data->pInRateTensor[i] != 0)
+            scale_ratio = data->pOutRateTensor[i] / static_cast<float>(data->pInRateTensor[i]);
+        else
+            scale_ratio = 0;
```

Use it as a ternary operator here
 ```
scale_ratio = (data->pInRateTensor[i] != 0) ? (data->pOutRateTensor[i] / static_cast<float>(data->pInRateTensor[i])) : 0;
```
Ideally if scale_ratio is 0, we need to throw an error